lib/phlex/sgml.rb

```diff
@@ -55,24 +55,34 @@ def element_method?(method_name)
 
 		# @abstract Override to define a template for your component.
 		# @example
-		# 	def template
+		# 	def view_template
 		# 		h1 { "👋 Hello World!" }
 		# 	end
 		# @example Your template may yield a content block.
-		# 	def template
+		# 	def view_template
 		# 		main {
 		# 			h1 { "Hello World" }
 		# 			yield
 		# 		}
 		# 	end
 		# @example Alternatively, you can delegate the content block to an element.
-		# 	def template(&block)
+		# 	def view_template(&block)
 		# 		article(class: "card", &block)
 		# 	end
 		def template
 			yield
 		end
 
+		def self.method_added(method_name)
+			if method_name == :template
+				Kernel.warn "Defining the `template` method on a Phlex component is deprecated and will be unsupported in Phlex 2.0. Please define `view_template` instead."
```

Comment:
:bulb: Seems I'm late to party with my story, since this check was already removed. I had to spent some time to find out which component emits this warning. If possible, add also some pointer like `self.class` to the output for future warning. :pray: I was able to find out just by opening gem with bundle open and adding debug print in this method temporarily in the end.

Reply:
Hey @simi, thanks for the feedback. Yes, we removed it from `main`, but it’s still in the latest version. I’ll try to remember to add more context to deprecation warnings in the future.

I hope we won’t need to do anymore of these going forward.